src/goto-instrument/contracts/contracts.cpp

```diff
@@ -309,7 +309,13 @@ void code_contractst::check_apply_loop_contracts(
 
   // Perform write set instrumentation on the entire loop.
   check_frame_conditions(
-    function_name, goto_function.body, loop_head, loop_end, loop_assigns);
+    function_name,
+    goto_function.body,
+    loop_head,
+    loop_end,
+    loop_assigns,
+    // do not skip checks on function parameter assignments
+    false);
```

Comment:
If we see a need to comment on the Boolean value then it may be better to introduce an `enum` that has informative values. If we always pass constant values, then it might be even better to use a template parameter.